### PR TITLE
Aiven

### PR DESCRIPTION
**Pull request by laurence-bird**

### CODE REVIEW

File: src/main/resources/prd.conf [hunk 1]
```diff
@@ -2,4 +2,23 @@ include "common.conf"
...
+  }
+  ssl {
+    enabled = true
+    keystore.location = "conf/client.keystore.p12"
+    keystore.password = "@@{prd.aiven.keystore.password}"
+    key.password = ${kafka.ssl.keystore.password}
+    truststore.location = "conf/client.truststore.jks"
+    truststore.password = "@@{prd.aiven.truststore.password}"
+  }
+  enableConsumers = true
```
Comment (fleetwoods) [inline src/main/resources/prd.conf]: What is this?

File: src/main/resources/prd.conf [hunk 2]
```diff
@@ -2,4 +2,23 @@ include "common.conf"
...
+  group.id = "audit-log"
```
Comment (fleetwoods) [inline src/main/resources/prd.conf]: Is this not in common?

Comment (laurence-bird) [inline src/main/resources/prd.conf]: correct, removed.

File: src/main/resources/prd.conf (hunk 1 above)
Comment (laurence-bird) [inline src/main/resources/prd.conf]: this is redundant, have removed

File: build.sbt [hunk 3]
```diff
@@ -1,51 +1,56 @@
 name := "composer"
 organization := "com.ovoenergy"
 
-scalaVersion := "2.11.8"
+scalaVersion := "2.11.9"
```
Comment (cb372) [inline build.sbt]: doesn't really matter, but latest is 2.11.11

File: cloudformation.yaml [hunk 4]
```diff
@@ -49,3 +49,14 @@ Resources:
...
+        -
+          PolicyName: read-aiven-bucket
+          PolicyDocument:
+            Statement:
+            - Effect: Allow
+              Action:
+                - 's3:Get*'
+                - 's3:List*'
+              Resource:
+                - 'arn:aws:s3:::ovo-comms-aiven'
```
Comment (cb372) [inline cloudformation.yaml]: Don't forget to update the prod stack before merging

File: src/servicetest/scala/servicetest/AivenServiceTestIt.scala [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+  val kafkaHosts = "localhost:29092"
+  val zkHosts = "localhost:32181"
+  val s3Endpoint = "http://localhost:4569"
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    initialiseConsumers()
+    uploadTemplateToS3()
+    Thread.sleep(30000L) // yeah this one will definitely fix everything
```
Comment (cb372) [inline src/servicetest/scala/servicetest/AivenServiceTestIt.scala]: could we get rid mega-sleep of this now we are not using docker-compose?

File: src/servicetest/scala/servicetest/AivenServiceTestIt.scala [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
```
Comment (cb372) [inline src/servicetest/scala/servicetest/AivenServiceTestIt.scala]: can delete this override, as it's just calling super

File: src/servicetest/scala/servicetest/DockerIntegrationTest.scala [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+    .withEnv(
+      "SCHEMA_REGISTRY_HOST_NAME=schema-registry",
+      "SCHEMA_REGISTRY_KAFKASTORE_CONNECTION_URL=aivenZookeeper:32182",
+      s"SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS=PLAINTEXT://$hostIpAddress:29093"
+    )
+    .withLogWritingAndReadyChecker("Server started, listening for requests", "schema-registry")
+
+  lazy val fakes3 = DockerContainer("lphoward/fake-s3:latest", name = Some("fakes3"))
+    .withPorts(4569 -> Some(4569))
+    .withReadyChecker(LogOutputAndWaitForLineThatContains("WEBrick::HTTPServer#start", "fakes3"))
```
Comment (cb372) [inline src/servicetest/scala/servicetest/DockerIntegrationTest.scala]: Looks like you forgot to switch this one to `withLogWritingAndReadyChecker`. Once you've done that you can remove the `LogOutputAndWaitForLineThatContains` class at the top of the file.

File: src/servicetest/scala/servicetest/LegacyServiceTestIT.scala [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+
+  val kafkaHosts = "localhost:29092"
+  val zkHosts = "localhost:32181"
+  val s3Endpoint = "http://localhost:4569"
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    initialiseConsumers()
+    uploadTemplateToS3()
+    Thread.sleep(30000L) // yeah this one will definitely fix everything
```
Comment (cb372) [inline src/servicetest/scala/servicetest/LegacyServiceTestIT.scala]: same comments as in the Aiven service test

File: src/servicetest/scala/servicetest/AivenServiceTestIt.scala [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+//Implicits
+import com.ovoenergy.comms.serialisation.Serialisation._
+
+class AivenServiceTestIt
```
Comment (cb372) [inline src/servicetest/scala/servicetest/AivenServiceTestIt.scala]: Can we stop putting "it" on the end of the test class name?

File: cloudformation.yaml (hunk 4 above)
Comment (laurence-bird) [inline cloudformation.yaml]: Yup thats been updated :)

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 192](https://circleci.com/gh/ovotech/comms-composer/192) at 2017-07-10 10:19 BST.

**Pull request merged**